### PR TITLE
Tune and add breakpoints for the navbar to address overflow issues under certain conditions.

### LINKED ISSUE #1205: Broken toolbar layout with long crate names + 'go to latest version'
Long-ish crate names, especially combined with the presence of the 'go to latest version' link, breaks the toolbar layout:

docsrs_toolbar

### PR DESCRIPTION
**Pull request by arusahni**
With crates with long names, and when the "Go to latest version" or "Yanked" warning badges present, the navbar doesn't overflow correctly.  This PR adds some additional breakpoints to ensure we don't experience the visual glitch of the overflow, as well as moves the "Source" link to the crate menu to buy some additional space.

## To test

```
# add two versions of a crate with a long name
cargo run -- build crate air-interpreter-wasm 0.0.33                                                                                                                                                                   
cargo run -- build crate air-interpreter-wasm 0.0.35

# Start the webserver
cargo run -- start-web-server --reload-templates

# From the web UI, verify that the breakpoints are correctly hit for the latest version
<browse to http://localhost:3000/air-interpreter-wasm/0.0.35/air_interpreter_wasm/index.html >
# From the web UI, verify that the breakpoints are correctly hit for an older version
<browse to http://localhost:3000/air-interpreter-wasm/0.0.33/air_interpreter_wasm/index.html >

# Now, yank both versions
psql postgresql://cratesfyi:password@localhost:15432 -c "UPDATE releases SET yanked = true FROM  crates WHERE releases.crate_id = crates.id AND crates.name = 'air-interpreter-wasm'"

# From the web UI, verify that the breakpoints are correctly hit for the latest version
<browse to http://localhost:3000/air-interpreter-wasm/0.0.35/air_interpreter_wasm/index.html >
# From the web UI, verify that the breakpoints are correctly hit for an older version
<browse to http://localhost:3000/air-interpreter-wasm/0.0.33/air_interpreter_wasm/index.html >
```

Fixes #1205

### CODE REVIEW

**Review by jyn514 — COMMENTED** [review]
I think this is the right approach but I don't feel qualified to review the implementation.

r? @GuillaumeGomez

File: templates/header/topbar_begin.html [hunk 1]
```diff
@@ -10,7 +10,9 @@
     <div class="container">
         <div class="pure-menu pure-menu-horizontal" role="navigation" aria-label="Main navigation">
             <form action="/releases/search" method="GET"
-                class="landing-search-form-nav {% if is_latest_version is defined and not is_latest_version %}not-latest{% endif %}">
+                                            class="landing-search-form-nav {%
+                                            if is_latest_version is defined and not is_latest_version %}not-latest{% endif
+                                            %} {% if metadata.yanked %}yanked{% endif %}">
```
[the remark this answers is not in the document]
Comment (arusahni) [reply templates/header/topbar_begin.html]: Good catch! Fixed.

File: templates/header/topbar_begin.html [hunk 2]
```diff
@@ -9,8 +9,11 @@
 <div class="nav-container">
     <div class="container">
         <div class="pure-menu pure-menu-horizontal" role="navigation" aria-label="Main navigation">
-            <form action="/releases/search" method="GET"
-                class="landing-search-form-nav {% if is_latest_version is defined and not is_latest_version %}not-latest{% endif %}">
+            <form action="/releases/search"
+                  method="GET"
```
Comment (GuillaumeGomez) [inline templates/header/topbar_begin.html:13]: Why did you put it on another line?

File: templates/header/topbar_begin.html [hunk 3]
```diff
@@ -9,8 +9,11 @@
...
+                  class="landing-search-form-nav {%
+                  if is_latest_version is defined and not is_latest_version %}not-latest{% endif
+                  %} {% if metadata.yanked %}yanked{% endif %}">
```
Comment (GuillaumeGomez) [inline templates/header/topbar_begin.html:16]: Is this case possible? Can we see the docs of a yanked crate? :o

File: templates/rustdoc/topbar.html [hunk 4]
```diff
@@ -84,6 +84,14 @@
                                 </a>
                             </li>
 
+                            {# A link to the release's source view #}
+                            <li class="pure-menu-item">
```
Comment (GuillaumeGomez) [inline templates/rustdoc/topbar.html:88]: Why did you move this item here?

File: templates/rustdoc/topbar.html [hunk 5]
```diff
@@ -84,6 +84,14 @@
...
+                                <a href="{{ crate_url | safe }}/source/" title="Browse source of {{ metadata.name }}-{{ metadata.version }}" class="pure-menu-link">
+                                    {{ "folder-open" | fas(fw=true) }}
```
Comment (GuillaumeGomez) [inline templates/rustdoc/topbar.html:90]: I'm curious about this: why change `far` to `fas(fw=true)`? What does it change?

File: templates/style/_navbar.scss [hunk 6]
```diff
@@ -131,38 +131,86 @@ div.nav-container {
...
-        @media screen and (min-width: 965px) {
+        display: none;
+        @media screen and (min-width: #{$medium-width}) {
             display: inline;
         }
     }
 
-    form.landing-search-form-nav.not-latest {
-        span.title {
+     /// Generate the rules for the navbar menu item labels at various dimensions.
```
Comment (GuillaumeGomez) [inline templates/style/_navbar.scss:155]: That's a bit weird to see `/// ` used in a scss file. Is it a typo or a feature of scss? (and if it's a feature, what does it do?)

**Comment by GuillaumeGomez** [comment]
Except for the nits, the code looks good to me (as far as I understand it). I didn't test it locally yet though. I'll try to whenever I have some time.

File: templates/header/topbar_begin.html (hunk 3 above)
Comment (Nemo157) [reply templates/header/topbar_begin.html:16]: Yes, since you can still depend on yanked crates and use them, we still allow accessing the docs.

Comment (GuillaumeGomez) [reply templates/header/topbar_begin.html:16]: I didn't know you could still use yanked crates. TIL.

File: templates/header/topbar_begin.html (hunk 2 above)
Comment (arusahni) [reply templates/header/topbar_begin.html:13]: When wrapping a tag, I generally prefer to place each attribute on their own line because it can be harder to visually locate the attribute otherwise. If this is something you'd rather avoid in the markup, I'm fine restoring it :+1:

File: templates/rustdoc/topbar.html (hunk 4 above)
Comment (arusahni) [reply templates/rustdoc/topbar.html:88]: In the discussion in #1205 , we realized that there was a need for some additional space in the navbar to accommodate the combination of long crate names and warnings. The "Source" menu item was identified as one to relocate into the crate menu.

File: templates/header/topbar_begin.html (hunk 2 above)
Comment (GuillaumeGomez) [reply templates/header/topbar_begin.html:13]: No personal preference here, just wondering. :)

File: templates/rustdoc/topbar.html (hunk 5 above)
Comment (arusahni) [reply templates/rustdoc/topbar.html:90]: It changes the icon style. Since the "Source" menu item is being relocated to the menu, the original icon didn't match its new siblings in fill or alignment. By switching to `fas`, it's now a solid filled icon. By adding `fw=true`, it's set to be full-width so it is visually aligned with the icons above and below.

Here's a screenshot of the final output to demonstrate the consistency.
Screen Shot 2020-12-28 at 21 54 14

File: templates/style/_navbar.scss (hunk 6 above)
Comment (arusahni) [reply templates/style/_navbar.scss:155]: [It's a Sass feature - it supports JSDoc-style comments.](https://sass-lang.com/documentation/syntax/comments#documentation-comments)

**Comment by arusahni** [comment]
@GuillaumeGomez thanks for your time! Looking forward to further review :+1:

**Comment by jyn514** [comment]
> I didn't test it locally yet though. I'll try to whenever I have some time.

I tested it, it works very well :)

File: templates/style/_navbar.scss (hunk 6 above)
Comment (GuillaumeGomez) [reply templates/style/_navbar.scss:155]: Oh ok. Not sure if it'll be useful here but I guess it doesn't hurt.

File: templates/rustdoc/topbar.html (hunk 5 above)
Comment (GuillaumeGomez) [reply templates/rustdoc/topbar.html:90]: Nice!

**Comment by GuillaumeGomez** [comment]
Looks good to me as well. Let's maybe wait for @Nemo157 to take a look as well?

**Comment by jyn514** [comment]
Nah, I think this is fine as is. Thanks for reviewing!
